src/components/HabitTracker/AddEditHabitModal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useRef } from "react";
+import { Button, Modal, Form } from "react-bootstrap";
+import { StyledModal } from "../styled/Modal";
+import { Habit } from "../../pages/habittracker/Habits";
+import { colors } from "../../common/color-utils";
+import styled from "styled-components";
+
+interface AddEditHabitModalProps {
+  show: boolean;
+  handleClose: () => void;
+  handleSaveHabit: (habit: Habit) => void;
+  originalHabit: Habit;
+  newHabit: Habit;
+  setNewHabit: (habit: Habit) => void;
+}
+
+// Styled component for the Save Button
+const SaveButton = styled.button<{ disabled: boolean }>`
+  background-color: ${({ disabled }) =>
+    disabled ? colors.disabledBgGrey : colors.blue};
+  border-color: ${({ disabled }) =>
+    disabled ? colors.disabledBgGrey : colors.blue};
+  color: white;
+  padding: 8px 16px;
+  border-radius: 5px;
+  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
+  transition: background-color 0.3s ease, border-color 0.3s ease;
+
+  &:hover {
+    background-color: ${({ disabled }) =>
+      disabled ? colors.disabledBgGrey : colors.bgWhite};
+    border-color: ${({ disabled }) =>
+      disabled ? colors.disabledBgGrey : colors.blue};
+    color: ${({ disabled }) =>
+      disabled ? colors.disabledBgGrey : colors.blue};
+  }
+`;
+
+export function AddEditHabitModal({
+  show,
+  handleClose,
+  handleSaveHabit,
+  originalHabit,
+  newHabit,
+  setNewHabit,
+}: AddEditHabitModalProps) {
+  // Ref for the habit name input
+  const habitNameRef = useRef<HTMLInputElement>(null);
```

Comment:
if you are using a `useRef` hook; make sure you understand how this is used. There are multiple use-cases for this hook. What you have here is correct.

Write down the list of questions anytime you are working on the code that you want to ask me next time.